resolver/resolver_test.go

```diff
@@ -417,27 +417,68 @@ func TestMultiError(t *testing.T) {
 	}
 }
 
-func TestTruncate(t *testing.T) {
-	tm := *newTruncated()
-	if !tm.Truncated {
+func TestTruncateSetTruncateBit(t *testing.T) {
+	testTruncateSetTruncateBit(t, dns.MinMsgSize)
+}
+
+func TestTruncateEdns0SetTruncateBit(t *testing.T) {
+	testTruncateSetTruncateBit(t, 4096)
+}
+
+func TestTruncateNoSetTruncateBit(t *testing.T) {
+	testTruncateNoSetTruncateBit(t, dns.MinMsgSize)
+}
+
+func TestTruncateEdns0NoSetTruncateBit(t *testing.T) {
+	testTruncateNoSetTruncateBit(t, 4096)
+}
+
+func testTruncateSetTruncateBit(t *testing.T, maxSize uint16) {
+	orig := *newMessage(maxSize)
```

Comment:
As you read this code - keep in mind where you have values and where you have pointers.

If `a` is a value, then `b := a` is a new value. Obviously for pointers (and reference types like slices and maps) `b` would now be pointing at the same value.